template/src/tests/__mocks__/browserMocks.js

```diff
@@ -0,0 +1,21 @@
+// Mock Browser API's which are not supported by JSDOM, e.g. ServiceWorker, LocalStorage
```

Comment:
Thought: maybe instead of jsdom we could run with `mock-browser`?
https://www.npmjs.com/package/mock-browser

Reply:
If you're trying to avoid jsdom, it's still using it under the hood. But yes, would be a nicer way to mock the full browser at once.

Reply:
it really depends how deep someone wants to test their API, e.g. testing serviceWorker basically means testing an Ajax call, for that one would mock that return anyway with Jest Mock functions. Sounds like a nice task / issue as next step 😄